tests/run-make/no-alloc-shim/rmake.rs

```diff
@@ -0,0 +1,40 @@
+// This test checks the compatibility of the interaction between `--emit obj` and
+// `#[global_allocator]`, as it is now possible to invoke the latter without the
+// allocator shim since #86844. As this feature is unstable, it should fail if
+// --cfg check_feature_gate is passed.
+// See https://github.com/rust-lang/rust/pull/86844
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+
+//@ ignore-msvc
+//FIXME(Oneirical): Getting this to work on MSVC requires passing libcmt.lib to CC,
+// which is not trivial to do.
+// Tracking issue: https://github.com/rust-lang/rust/issues/128602
+// Discussion: https://github.com/rust-lang/rust/pull/128407#discussion_r1702439172
+
+use run_make_support::{cc, cwd, has_extension, has_prefix, run, rustc, shallow_find_files};
+
+fn main() {
+    rustc().input("foo.rs").crate_type("bin").emit("obj").panic("abort").run();
+    let libdir = rustc().print("target-libdir").run().stdout_utf8();
+    let libdir = libdir.trim();
+    let alloc_libs = shallow_find_files(&libdir, |path| {
+        has_prefix(path, "liballoc-") && has_extension(path, "rlib")
+    });
+    let core_libs = shallow_find_files(libdir, |path| {
+        has_prefix(path, "libcore-") && has_extension(path, "rlib")
+    });
```

Comment:
Problem: as I mentioned in https://github.com/rust-lang/rust/pull/128407#discussion_r1703463892 (I know it's a bit buried among other review comments), this also needs to link in `libcompiler_builtins`:

```rs
let compiler_builtins_libs = shallow_find_files(libdir, |path| {
    has_prefix(path, "libcompiler_builtins") && has_extension(path, "rlib")
});
```